packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/http/invoke.ts

```diff
@@ -0,0 +1,180 @@
+  /**
+   * Encode arrays using commas. For example, {"array": ["a","b","c"]} encodes to "array=a,b,c,d"
+   */
+  COMMAS = 'COMMAS',
+  /**
+   * Encode arrays using the index value. For example, {"array": ["a","b","c"]} encodes to "array[0]=a&array[1]=b&array[2]=c"
+   */
+  INDICES = 'INDICES',
+  /**
+   * Repeat key for each item in the array. For example, {"array": ["a","b","c"]} encodes to "array[]=a&array[]=b&array[]=c"
+   */
+  REPEAT = 'REPEAT',
+}
+
+/**
+ * Properties for calling an external HTTP endpoint with HttpInvoke.
+ */
+export interface HttpInvokeProps extends sfn.TaskStateBaseProps {
+  /**
+   * Permissions are granted to call all resources under this path.
+   * @example "https://api.example.com"
+   *
+   */
+  readonly apiRoot: string;
+
+  /**
+   * The API endpoint to call, relative to `apiRoot`.
+   * @example sfn.TaskInput.fromText("path/to/resource")
+   *
+   */
+  readonly apiEndpoint: sfn.TaskInput;
+
+  /**
+   * The HTTP method to use.
+   * @example sfn.TaskInput.fromText("GET")
+   *
+   */
+  readonly method: sfn.TaskInput;
+
+  /**
+   * The EventBridge Connection to use for authentication.
+   *
+   */
+  readonly connection: events.IConnection;
+
+  /**
+   * The body to send to the HTTP endpoint.
+   *
+   * @default - No body is sent with the request.
+   */
+  readonly body?: sfn.TaskInput;
+
+  /**
+   * The headers to send to the HTTP endpoint.
+   *
+   * @default - No additional headers are added to the request.
+   * @example sfn.TaskInput.fromObject({ 'Content-Type': 'application/json' })
+   */
+  readonly headers?: sfn.TaskInput;
+
+  /**
+   * The query string parameters to send to the HTTP endpoint.
+   *
+   * @default - No query string parameters are sent in the request.
+   */
+  readonly queryStringParameters?: sfn.TaskInput;
+
+  /**
+   * When `true`, the HTTP request body is the URL-encoded form data of the `RequestBody` field.
+   * When `false` (default), the HTTP request body is the JSON-serialized `RequestBody` field.
+   * If set to `true`, also sets 'content-type' header to 'application/x-www-form-urlencoded'.
+   *
+   * @default - No encoding.
+   */
+  readonly urlEncodeBody?: boolean;
+
+  /**
+   * The format of the array encoding.
+   * Only used if `urlEncodeBody` is `true`.
+   *
+   * @default - ArrayEncodingFormat.INDICES
+   */
+  readonly arrayEncodingFormat?: ArrayEncodingFormat;
+}
+
+/**
+ * A Step Functions Task to call a public third-party API.
+ *
+ */
+export class HttpInvoke extends sfn.TaskStateBase {
+  protected readonly taskMetrics?: sfn.TaskMetricsConfig;
+  protected readonly taskPolicies?: iam.PolicyStatement[];
+
+  constructor(
+    scope: Construct,
+    id: string,
+    private readonly props: HttpInvokeProps,
+  ) {
+    super(scope, id, props);
+
+    this.taskPolicies = this.buildTaskPolicyStatements();
+  }
+
+  /**
+   * Provides the HTTP Invoke service integration task configuration.
+  */
+  /**
+   * @internal
+  */
```

Comment:
nit: this formatting looks off

Reply:
Thanks for pointing these out, I obviously didn't pay enough attention to the comments (and don't normally use them). I did a sweep and they should all be consistent now.